code/examples/main-can_bus_demo.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+
+This main is used to test the module-scheduler
+For this programm to compile, the supplied Makefile must be used
+
+This program makes the can-bus demo run on one arduino using the
+module-scheduler
+
+for the button, I/O pin 53 should be used, there is no external pullup resistor
+needed. by pushing the button, the built-in led should turn on, by releasing the
+button, the led should turn off.
+
+*/
+
+#include <hwlib.hpp>
+#include <rtos.hpp>
+
+#include <can_bus_pending_task.hpp>
+#include <module_task.hpp>
+
+// include the classes from the can-bus demo, located in the
+// internal-communication folder
+#include <button/module.hpp>
+#include <controller/module.hpp>
+#include <led/module.hpp>
+
+namespace r2d2::module_scheduler {
+
+    // this class initializes the led-module task with the arguments needed by
+    // the module
+    class led_task_c : public module_task_c<r2d2::led::module_c> {
+    public:
+        // fitst initalzie the output pin, then initialize the module-task with
+        // this pin
+        // the output pin must be initialized using the allocator,
+        // because the pin must be initialized before the module-task
+        led_task_c(alloc_c &alloc)
+            : module_task_c(alloc.alloc_object<hwlib::target::pin_out>(
+                  hwlib::target::pins::led)) {
+        }
+    };
+
+    // this class initializes the controller-module task with the arguments
+    // needed by the module
+    class controller_task_c : public module_task_c<r2d2::controller::module_c> {
+    public:
+        controller_task_c(alloc_c &alloc) : module_task_c() {
+        }
+    };
+
+    // this class initializes the button-module task with the arguments
+    // needed by the module
+    class button_task_c : public module_task_c<r2d2::button::module_c> {
+    private:
+        /**
+         * Create a hwlib::target::pin_in, enable the pullup resistor and return
+         * the pin
+         *
+         */
+        static hwlib::target::pin_in &
+        allocate_pin_with_pullup(alloc_c &alloc, hwlib::target::pins pin) {
+            auto &pin_in = alloc.alloc_object<hwlib::target::pin_in>(pin);
+            pin_in.pullup_enable();
+            return pin_in;
+        }
+
+    public:
+        // fitst initalzie the input pin, then initialize the module-task with
```

Comment:
Spelling

Reply:
fixed in https://github.com/R2D2-2019/module_scheduler/pull/2/commits/83c4e9a3a83e98ef4faf6f67ca42d4a2c130eba3